src/profiler/profiler.h

```diff
@@ -1225,6 +1251,10 @@ struct ProfileOperator : public ProfileEvent {
   static ProfileDomain domain_;
   /*! \brief Optional operator attributes */
   std::unique_ptr<Attributes> attributes_;
+  /*! \brief Whether to profile as operator */
+  bool profiling_operator_;
```

Comment:
Can we make these two bool members const and initlialize them during the class initialization? The reason being that your start() and stop() functions depend on these two variables. If during the execution of the program these two variables are modified between start() and stop(), then stop() will generate undefined behavior.

Reply:
That's great advice! Now I am declaring them as const and using the the initializing list